[en]: update egressgateway/create-egpolicy (#5183)
DaoCloud/DaoCloud-docs: docs/en/docs/network/modules/egressgateway/create-egpolicy.md

# Create EgressPolicy

Egresspolicy is used to define which pods' outbound traffic needs to be forwarded through the
EgressGateway node, and to define other configuration details.
When a matching pod accesses any address outside the cluster (any address that is not a Node IP,
CNI Pod CIDR, or ClusterIP), it will be forwarded by the EgressGateway Node.
Currently, EgressPolicy is divided into two categories: namespace-level and cluster-level.

- __Namespace-level__ : The scope of the created policy is at the namespace level.
- __Cluster-level__ : The scope of the created policy is at the cluster level.

## Create Namespace EgressPolicy

> 1. Click on __Gateway Policy__ -> Create __Namespace Policy__ , and fill in the following parameters:

Reviewer: ```suggestion
1. Click __Gateway Policy__ -> __Create Namespace Policy__ , and fill in the following parameters:
```

    **Outbound Address**:

    * __Outbound IP Address__ : Use a separate VIP as the outbound IP. Since EgressGateway operates on the basis of ARP,
    it is suitable for traditional networks where the source IP is always fixed. If not set, the default VIP will be used,
    and the IP value must be within the IPPool range of the EgressGateway. You can choose the IP in two ways:

>         * __Specify Outbound IP Address__ : Specify a specific IP address as the outbound IP by [ selecting Gateway](./create_eg.md) proper Egress IP range.

Reviewer: ```suggestion
        * __Specify Outbound IP Address__ : Specify a specific IP address as the outbound IP by [selecting Gateway](./create_eg.md) from a proper Egress IP range.
```

    but the disadvantage is that the outbound source IP may change with node failures.
    The corresponding field information is `spec.egressIP.useNodeIP=true` .

    **Source Address Pods** :

    __Select Pods__ : Support selecting source address pods through __Label Selectors__ or __Source Address__ ,
    specifying the range of Pods where this policy will take effect.
    When a matching Pod accesses any address outside the cluster (any address that is not a Node IP,
    CNI Pod CIDR, or ClusterIP), it will be forwarded by the EgressGateway Node.

    - __Label Selectors__ : Specify the source address pods using labels.
    - __Source Address__ : By adding source address CIDR blocks to the whitelist, the locked pods will enforce this policy.

    **Advanced Settings** :

    __Destination Address__ : Specify a whitelist of destination addresses. When specified,
    this policy will only apply to the defined destination addresses. Supports input in various
    formats such as single IP address, IP range, CIDR, etc. Default is to apply to all destination addresses.

2. Click **OK** after inputting the details to complete the creation.

## Create Cluster EgressPolicy

> 1. Click on __Gateway Policy__ -> __Create Cluster Policy__ , and refer to the parameters for __Basic Information__, __Outbound Address__, and __pods Selection__ as described in [Create Namespace EgressPolicy](#create-namespace-egresspolicy).

Reviewer: ```suggestion
1. Click __Gateway Policy__ -> __Create Cluster Policy__ , and refer to the parameters for __Basic Information__, __Outbound Address__, and __Pod Selection__ as described in [Create Namespace EgressPolicy](#create-namespace-egresspolicy).
```
`Click on` 可以全局替换一下，参见  https://github.com/DaoCloud/DaoCloud-docs/issues/4956

    In the above creation command:

    * `spec.egressGatewayName` specifies which EgressGateway group to use.
    * `spec.appliedTo.podSelector` specifies which pods within the cluster this policy will apply to.
    * There are two options for the source IP address of egress traffic:

>         * You can use the IP of the gateway node. This is suitable for public cloud and traditional
>         network environments, but the drawback is that the outbound source IP may change with node failures.
>         You can set `spec.egressIP.useNodeIP=true` to make it effective.
>         * You can use a separate VIP. Since EgressGateway operates on the basis of ARP,
>         it is suitable for traditional networks, but not suitable for public cloud environments.
>         The advantage is that the outbound source IP is always fixed. If no settings are made in EgressPolicy,
>         the default VIP of `egressGatewayName` will be used, or you can manually specify `spec.egressIP.ipv4`,
>         and the IP value must be within the IP pool of the EgressGateway.

Reviewer: ```suggestion
        * You can use the IP of the gateway node. This is suitable for public cloud and traditional
          network environments, but the drawback is that the outbound source IP may change with node failures.
          You can set `spec.egressIP.useNodeIP=true` to make it effective.
        * You can use a separate VIP. Since EgressGateway operates on the basis of ARP,
          it is suitable for traditional networks, but not suitable for public cloud environments.
          The advantage is that the outbound source IP is always fixed. If no settings are made in EgressPolicy,
          the default VIP of `egressGatewayName` will be used, or you can manually specify `spec.egressIP.ipv4`,
          and the IP value must be within the IP pool of the EgressGateway.
```
折行之后也要对齐

## Egress Gateway Test

> 1. Deploy the nettools application outside the cluster to simulate an external service. nettools
> will return the requester's source IP address in the HTTP response.

Reviewer: ```suggestion
1. Deploy the nettools application outside the cluster to simulate an external service. nettools
   will return the requester's source IP address in the HTTP response.
```

    ```shell
    docker run -d --net=host ghcr.io/spidernet-io/egressgateway-nettools:latest /usr/bin/nettools-server -protocol web -webPort 8080
    ```

> 2. In the visitor Pod within the cluster, verify the effect of outbound traffic. When the visitor accesses
> the external service, the source IP returned by nettools should match the effect of EgressPolicy `.status.eip`.

Reviewer: ```suggestion
2. In the visitor Pod within the cluster, verify the effect of outbound traffic. When the visitor accesses
   the external service, the source IP returned by nettools should match the effect of EgressPolicy `.status.eip`.
```
